lib/settings.py

```diff
@@ -212,47 +190,50 @@ def prompt(question, choices):
+            LOGGER.info("Beginning download..")
+            with open("{}.txt".format(filename), "a+") as wordlist:
+                response = requests.get(base64.b64decode(b64link.strip()), stream=True)
+                total_length = response.headers.get('content-length')
                 if verbose:
-                    LOGGER.debug("Starting download at {}..".format(start))
-                downloaded = 0
-                total_length = int(total_length)
-                for data in response.iter_content(chunk_size=1024):
-                    downloaded += len(data)
-                    wordlist.write(data)
-                    done = int(50 * downloaded / total_length)
-                    sys.stdout.write("\r[\033[93m{}\033[0m{}]".format("#" * done, " " * (50-done)))
-                    sys.stdout.flush()
-        print("")
-        LOGGER.info("Download complete, saved under: {}.txt. Time elapsed: {}s".format(filename, time.time() - start))
-    else:
-        if multi <= len(WORDLIST_LINKS):
-            for _ in range(int(multi)):
-                LOGGER.info("Downloading wordlist #{}..".format(_ + 1))
-                download_rand_wordlist(verbose=verbose)
+                    LOGGER.debug("Content length to be downloaded: {}..".format(convert_file_size(int(total_length))))
+                    LOGGER.debug("Wordlist link downloading from: '{}'..".format(b64link.strip()))
+                if total_length is None:
+                    wordlist.write(response.content)
+                else:
+                    start = time.time()
+                    if verbose:
+                        LOGGER.debug("Starting download at {}..".format(start))
+                    downloaded = 0
+                    total_length = int(total_length)
+                    for data in response.iter_content(chunk_size=1024):
+                        downloaded += len(data)
+                        wordlist.write(data)
+                        done = int(50 * downloaded / total_length)
+                        sys.stdout.write("\r[\033[93m{}\033[0m{}]".format("#" * done, " " * (50-done)))
+                        sys.stdout.flush()
+            print("")
+            LOGGER.info("Download complete, saved under: {}.txt. Time elapsed: {}s".format(filename, time.time() - start))
         else:
-            LOGGER.fatal(
-                "There are currently {} wordlists available for download, and you "
-                "have chose to download {}. Seeing as there aren't that many right now "
-                "this download will fail, try again with a smaller number..".format(len(WORDLIST_LINKS), multi)
-            )
+            if multi <= len(wordlist_links.readlines()):
+                for _ in range(int(multi)):
+                    LOGGER.info("Downloading wordlist #{}..".format(_ + 1))
+                    download_rand_wordlist(verbose=verbose)
+            else:
+                wordlist_links.seek(0)
```

Comment:
Because if you read the list it will stop at the end, so if you don't have `seek(0)` it will say there's nothing there. It's a weird Python thing that I've come across a few times.